slsim/Sources/source.py

```diff
@@ -123,3 +123,53 @@ def extended_source_magnitude(self, band):
             band_string = "mag_" + band
         source_mag = self.source_dict[band_string]
         return source_mag
+
+    def extended_source_position(self, deflector_center, test_area):
+        """Extended source position. If not present from the catalog, it is drawn
+        uniformly within the circle of the test area centered on the deflector position.
+
+        :param deflector_center: center of the deflector.
```

Comment:
doc string for 'test_area' missing. Perhaps also re-name this variable for this function, as conceptually in this function, it has a different purpose.